mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/auth/AuthenticationMode.kt

```diff
@@ -0,0 +1,11 @@
+package ch.epfl.sdp.mobile.infrastructure.persistence.auth
+
+/** An enumeration representing the [AuthenticationMode] that a user chose. */
+enum class AuthenticationMode {
+
+  /** The mode in which a user wants to login to their existing account. */
+  SignIn,
+
+  /** The mode in which a user wants to create a new account. */
+  SignUp,
+}
```

Comment:
It looks like this `AuthenticationMode` is an implementation detail which, while useful when using the `authenticate` method, is not relevant in the `persistence.auth` package.

Maybe make it private inside the `FirebaseAuth` class ?

Reply:
Fully agree, fixed in 12d715c